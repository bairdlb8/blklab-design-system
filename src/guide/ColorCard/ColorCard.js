import React from 'react'
import PropTypes from 'prop-types'

import './ColorCard.scss'

const ColorCard = ({ className = '', hexColor = '', name = '', usage = '' }) => {
  console.log(hexColor)
  return (
    <div className="ds-color-card">
      <div className="ds-color-card__color" style={{ '--ds-swatch-color': hexColor }}></div>
      <div className="py-sm px-sm">
        <b>Name:</b> {name}
        <br />
        <b>Token:</b> {usage}
        <br />
        <b>Hex:</b> {hexColor}
      </div>
    </div>
  )
}

ColorCard.propTypes = {}

export default ColorCard
