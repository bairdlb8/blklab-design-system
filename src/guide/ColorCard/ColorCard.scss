.ds-color-card {
  --ds-swatch-color: #000;

  background-color: #fff;
  box-shadow: var(--shadow);

  &__color {
    background-color: var(--ds-swatch-color);
    height: 200px;
  }

  &__description {
  }
}
