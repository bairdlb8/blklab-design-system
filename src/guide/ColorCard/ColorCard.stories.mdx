import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import ColorCard from './ColorCard'

<Meta title="Guide / ColorCard" component={ ColorCard } />

# ColorCard

## Add some text here

export const Template = (args) => {
  return <ColorCard {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={ {} }>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={ ColorCard } />
