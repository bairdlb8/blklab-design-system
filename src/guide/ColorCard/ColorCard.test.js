import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import ColorCard from './ColorCard'

expect.extend(toHaveNoViolations)

describe('<ColorCard />', () => {
  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<ColorCard />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<ColorCard />)

    expect(asFragment(<ColorCard />)).toMatchSnapshot()
  })
})
