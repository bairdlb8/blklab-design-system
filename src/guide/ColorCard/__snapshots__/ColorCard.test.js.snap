// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorCard /> should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    ColorCard
  </div>
</DocumentFragment>
`;
